test/no-mod-iter-var-for-loop.js

```diff
@@ -0,0 +1,57 @@
+/**
+ * @fileoverview Tests for no-mod-iter-var-for-loop rule
+ * @author Nicolas Feignon <nfeignon@gmail.com>
+ */
+
+'use strict';
+
+var Solium = require ('solium');
+var wrappers = require("./utils/wrappers");
+var toContract = wrappers.toContract;
+
+var userConfig = {
+    rules: {
+        "security/no-mod-iter-var-for-loop": "error"
+    }
+};
+
+describe ('[RULE] no-mod-iter-var-for-loop: Acceptances', function () {
+
+    it ('should accept for loops that don\'t modifify their iteration variable', function (done) {
+        var code = [
+            'function foo() { for(i = 0; i < 10; i++) { i + 1; } }',
+            'function foo() { for(;;) {} }',
+            'function foo() { for(a;;) { a++; } }',
+        ];
+        var errors;
+
+        for (let e of code) {
+            errors = Solium.lint (toContract(e), userConfig);
+            errors.length.should.equal (0);
+        }
+
+        Solium.reset ();
+        done ();
+    });
+});
+
+
+describe ('[RULE] no-mod-iter-var-for-loop: Rejections', function () {
+
+    it ('should reject for loops that modify their iteration variable', function (done) {
+        var code = [
+            'function foo () { for(i = 0; i < 10; i++) { i = 5; } }',
+            'function foo () { for(i = 0; i < 10; i++) { i++; } }',
+            'function foo () { for(i = 0; i < 10; i++) { i += 1; } }',
+        ];
+        var errors;
+
+        for (let e of code) {
+            errors = Solium.lint (toContract(e), userConfig);
+            errors.length.should.be.above (0);
```

Comment:
Don't test for "above 0". Instead have a test like `errors.should.have.size(<EXACT NUMBER OF ERRORS PRODUCED BY THS TESTED CODE>)`.
This means you might have to remove the for loop and test each code in the array separately, but that is the better practice.
We need to ensure that **given line produces exactly N amount of errors when only this rule is being applied**.